### PR TITLE
Add warmth to status line

### PR DESCRIPTION
**Pull request by zwim**
This PR adds the possibility to add a frontlight warmth indicator to the bottom status bar.

Tested on Android (mobile and Tolino) and with the emulator.
Not tested on Kobos, Cervantes, Pocketbook. Could/wants someone to check these please.

grafik

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/8060)

### CODE REVIEW

File: frontend/apps/reader/modules/readerfooter.lua
Comment (NiLuJe) [inline frontend/apps/reader/modules/readerfooter.lua:46]: Please don't modify existing values, I vaguely remember discussing this in the past, we allowed it, and it came back to bite us in the ass later ;).

File: frontend/apps/reader/modules/readerfooter.lua
Comment (NiLuJe) [inline frontend/apps/reader/modules/readerfooter.lua:159]: Does the "stub" variant of `getWarmth` return 0? because if it returns `nil`, this'll probably crash ;).

File: frontend/device/kobo/powerd.lua
Comment (NiLuJe) [inline frontend/device/kobo/powerd.lua:269]: This probably ought to be a simple `hasNaturalLight` check [when in doubt, check what the FrontLight widget does, it should handle that distinction properly].

Comment (zwim) [reply frontend/device/kobo/powerd.lua:269]: I trust you :) But it should be checked on a Kobo with and on an Kobo without naturalLight.

Comment (NiLuJe) [reply frontend/device/kobo/powerd.lua]: Yeah, I'll have some time to test it next Thursday if nobody got to it first ;).

**Review by poire-z — COMMENTED** [review]
Looks OK, no real comment. But as:

> Not tested on Kobos, Cervantes, Pocketbook. Could/wants someone to check these please.

Safer to merge after v2021.08 is released.

**Comment by zwim** [comment]
Yes, let us wait after the 2021.8 release.

**Comment by NiLuJe** [comment]
(Will test later, hopefully this WE).

File: frontend/apps/reader/modules/readerfooter.lua
Comment (NiLuJe) [reply frontend/apps/reader/modules/readerfooter.lua:159]: The same applies to the Kobo branch, actually ;).

Comment (NiLuJe) [reply frontend/apps/reader/modules/readerfooter.lua:159]: Also, it's already a percent on Kobo/Cervantes, so, it probably ought to be displayed as such, like the FrontLight level ;). (i.e., `%d%%`).

**Review by NiLuJe — CHANGES_REQUESTED** [review]
Barring the crash potential on Kobo mentioned earlier, it works fine over here on devices w/ NL ;).

File: frontend/apps/reader/modules/readerfooter.lua
Comment (NiLuJe) [reply frontend/apps/reader/modules/readerfooter.lua:159]: Actually, I'd *definitely* make sure the `getWarmth` API is consistent and *always* returns a percentage: i.e., move the Android scaling to the Android powerd ;).

(c.., the get*HW vs. get* stuff powerd is riddled with for this kind of stuff).

Comment (zwim) [reply frontend/apps/reader/modules/readerfooter.lua:159]: > The same applies to the Kobo branch, actually ;).
> 
> (Because there are a number of Kobo w/o NaturalLight where it'll return nil ;).)

It seems something is wrong here with the displayed code (maybe caused by my `push force`): If you look under files changed you can see, that I have added a protector in Line 159
```
 elseif powerd:getWarmth() then
```

Comment (zwim) [reply frontend/apps/reader/modules/readerfooter.lua:159]: > Actually, I'd _definitely_ make sure the `getWarmth` API is consistent and _always_ returns a percentage: i.e., move the Android scaling to the Android powerd ;).
> 
> (c.., the get_HW vs. get_ stuff powerd is riddled with for this kind of stuff).

That is already done. I will change the display in `devicelistener` to percent, to be consistent.

File: frontend/apps/reader/modules/readerfooter.lua
Comment (NiLuJe) [inline frontend/apps/reader/modules/readerfooter.lua:157]: I was talking about that one ;).

This will return `nil` on Kobo/cervantes devices without amber leds ;). And that'll crash ;).

Comment (NiLuJe) [reply frontend/apps/reader/modules/readerfooter.lua:157]: i.e., I'd just leave a single branch, you can safely floor on every platform.

Something along the lines of

```lua
local warmth = powerd:getWarmth()
if warmth then
    return (prefix .. " %d%%"):format(math.floor(warmth))
end
```

Comment (NiLuJe) [reply frontend/apps/reader/modules/readerfooter.lua:157]: Actualy, you don't even need to floor *anywhere*.

printf("%d") does a cast, that truncates.

Comment (zwim) [reply frontend/apps/reader/modules/readerfooter.lua:157]: > This will return `nil` on Kobo/Cervantes devices without amber leds ;). And that'll crash ;).

Ohh, yes!

**Comment by poire-z** [comment]
There are 4 warmth related PRs waiting to be merged :)
I haven't followed much, so please say which of them are good to go / not obsoleted by some of the other PRs.

**Comment by zwim** [comment]
Here we go:
#8060 is waiting for @NiLuJe's approval. I hope his concerns are removed. Is not connected to the others.

#8129 is on the way. Nightmode tested, easy mode not ready tested, naming most probably 'Auto warmth'. Will be ready from my side on Sunday afternoon -> I expect some wording changes then.
#8154 sound ready to be merged. But I really suggest merging it together with #8129 as this would be the logical replacement.

#8120 is from @mergen3107: I can not tell the status. It is quite device specific. Not connected to the others.

**Comment by NiLuJe** [comment]
This should be mostly good to go, I haven't had time to do a final test, but the changes look good ;).

**Comment by mergen3107** [comment]
My PR is not ready yet :’) 
I don’t know when I will have time for it

**Pull request merged**
